extensions/wikia/WikiaWhiteList/WikiaWhiteList.php

```diff
@@ -93,6 +94,19 @@ function wfWhiteListParse($text) {
 							if (count($m) > 0) {
 								$text = preg_replace(sprintf(REGEX_PARSE_WHITELIST,"rel","nofollow"), "", $text);
 								$text = preg_replace(sprintf(CLASS_PARSE_WHITELIST,"external","(text|free|autonumber)"), "class=\"$1\"", $text);
+
+								$parsed = @parse_url( $lhref );
+								if (!empty( $parsed ) && is_array( $parsed )) {
+									$city_id = WikiFactory::DomainToID(wfNormalizeHost( $parsed['host'] ));
```

Comment:
this is the tricky part. for language wikis this will not work, as the city_url is a host and a language path (in case the first part of the url is a language code). For sure WikiFactoryLoader does similar check in https://github.com/Wikia/app/blob/dev/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L101-L111. Maybe there is a piece of code you could reuse for it.